### PR TITLE
Improve xtraarmors mod and remove IRC mods

### PR DESCRIPTION
**Pull request by LoneWolfHT**
Backwards compatibility testing needed before adding. (xtraarmors mod)

The IRC mod is probably better to add separately. Since not all server hosts may want it and it doesn't work on Linux anyway IIRC

I replaced the repetitive colored wool armor registrations with a for loop that takes colors from a table and creates a new set of armor for each color in that table.

This also fixes a bug in **3d_armor/3d_armor/api.lua** where there was an extra 'end'

### CODE REVIEW

**Comment by ClobberXD** [comment]
Just want to confirm: I don't see any code improvement to `xtraarmor`, is this PR still WIP? Have you renamed the textures for any specific reason?

I'm all for removing `irc` from the repo. :+1:

**Comment by LoneWolfHT** [comment]
> Just want to confirm: I don't see any code improvement to `xtraarmor`, is this PR still WIP? Have you renamed the textures for any specific reason

https://github.com/Billy-S/kingdoms_game/pull/30/commits/a16e4fa14a44eed18d6e569b72a6f56a1a60d337#diff-c14df6f48bf43703010177c1622644c4

It went from 1k lines to about 300 IIRC

**Comment by ClobberXD** [comment]
> [a16e4fa#diff-c14df6f48bf43703010177c1622644c4](https://github.com/Billy-S/kingdoms_game/commit/a16e4fa14a44eed18d6e569b72a6f56a1a60d337#diff-c14df6f48bf43703010177c1622644c4)
> 
> It went from 1k lines to about 300 IIRC

Oh, I didn't see that diff amidst all the texture renames, sorry. :)

Please add a short description to the first post, about what you've actually improved in `xtraarmor`.

File: mods/xtraarmor/init.lua [hunk 1]
```diff
@@ -12,997 +12,309 @@ minetest.register_craft({
...
+		inventory_image = "xtraarmor_inv_chestplate_wool_"..color..".png",
+		groups = {wool_chestplate=1, armor_torso=12,  armor_use=1000},
 		wear = 0,
 	})
-
-	minetest.register_tool("xtraarmor:chestplate_leather_dark_grey", {
-		description = "dark grey wool tunic",
-		inventory_image = "xtraarmor_inv_chestplate_leather_dark_grey.png",
-		groups = {leather_chestplate=1, armor_torso=12,  armor_use=1000},
+
```
Comment (ClobberXD) [inline mods/xtraarmor/init.lua]: Whitespace added to empty line.

File: mods/xtraarmor/init.lua [hunk 2]
```diff
@@ -12,997 +12,309 @@ minetest.register_craft({
...
+	local color = string.gsub(string.lower(colorname), ' ', '_')
 
-	minetest.register_tool("xtraarmor:leggings_leather_orange", {
-		description = "orange wool trousers",
-		inventory_image = "xtraarmor_inv_leggings_leather_orange.png",
-		groups = {leather_leggings=1, armor_legs=7,  armor_use=150},
+	minetest.register_tool("xtraarmor:helmet_wool_"..color, {
+		description = colorname.." Cap",
+		inventory_image = "xtraarmor_inv_helmet_wool_"..color..".png",
+		groups = {wool_helmet=1, armor_head=7,  armor_use=1000},
```
Comment (ClobberXD) [inline mods/xtraarmor/init.lua]: Spaces around all `..` and `=` symbols will improve readability.

File: mods/xtraarmor/init.lua [hunk 3]
```diff
@@ -12,997 +12,309 @@ minetest.register_craft({
...
 minetest.register_craft({
-	type = "shapeless",
-	output = "xtraarmor:helmet_leather_violet",
-	recipe = {"xtraarmor:helmet_leather", "dye:violet"},
+output = 'xtraarmor:helmet_chainmail',
+recipe = {
+	{'xpanes:bar', 'xpanes:bar', 'xpanes:bar'},
+	{'xpanes:bar', '', 'xpanes:bar'},
+	{'', '', ''},
+}
```
Comment (ClobberXD) [inline mods/xtraarmor/init.lua]: All lines in this craft def should be indented by one tab-space. Similarly, in other blocks of code with missing indents as well.

**Comment by LoneWolfHT** [comment]
Done
I also realized it's probably possible to use texture modifiers for the colored wool armor. So this PR is WIP

File: mods/xtraarmor/init.lua (hunk 1 above)
Comment (LoneWolfHT) [reply mods/xtraarmor/init.lua]: Fixed

File: mods/xtraarmor/init.lua (hunk 2 above)
Comment (LoneWolfHT) [reply mods/xtraarmor/init.lua]: Fixed

File: mods/xtraarmor/init.lua (hunk 3 above)
Comment (LoneWolfHT) [reply mods/xtraarmor/init.lua]: Whoops. I never got around to fixing that

**Comment by LoneWolfHT** [comment]
@ClobberXD You can look over it again.
I also fixed a bug in 3d_armor/3d_armor/api.lua where there was an extra 'end'

**Review by ClobberXD — APPROVED** [review]
LGTM, :+1: for the code. Please merge only after testing.

**Comment by LoneWolfHT** [comment]
I did some quick testing and it seems to work

**Comment by LoneWolfHT** [comment]
@Billy-S I've tested and found no problems.
 Although @ClobberXD Might want to check and make sure I didn't mess up anything when I removed the 3d_armor fix lol

**Comment by ClobberXD** [comment]
I think you should've revert  and , instead of adding another commit, since now your branch adds the same code that already exists and will result in a merge conflict. So now you have 3 commits to revert:

- 7ff9f4e4813fafef37e1c5326ed824f9acd5f639
- c06910209359a36e65f18d67f8b6dc66df5bd28a
- e6e98912c279ecc8662450eb3e75472e736f82f7

Tip: Use `git revert <commit-hash>`.  The order is important here, and the latest commits should be reverted first.

**Pull request merged**